client/src/components/Patients/index.js

```diff
@@ -1,32 +1,64 @@
 import React, { useState, useEffect } from 'react';
-import { Table, Typography, Input, message, Spin } from 'antd';
+import { Table, Typography, Input, message, Spin, Select } from 'antd';
 import axios from 'axios';
 import './style.css';
 import { useHistory } from 'react-router-dom';
+import { differenceInCalendarYears, parse } from 'date-fns';
 
 const { Search } = Input;
 const { Title } = Typography;
+const { Option } = Select;
 
 const Patients = () => {
   const [dataSource, setDataSource] = useState([]);
   const [loading, setLoading] = useState(true);
+  const [searchKey, setSearchKey] = useState('firstName');
   const history = useHistory();
 
-  const fetchPatients = async () => {
+  const buildUrlQuery = (params) => {
```

Comment:
I think you don't need to make a function to buildUlrQuery,
axios library already have an build option for that
```js
axios.get('/api/v1/patients/search',{params:{name:'alaa'}})
```